modules/react-components/src/tree-view/tree-view.tsx

```diff
@@ -0,0 +1,351 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FunctionComponent, ReactElement, useState, useEffect } from "react";
+import isNil from 'lodash/isNil';
+import isEmpty from 'lodash/isEmpty';
+import isEqual from 'lodash/isEqual';
+import find from 'lodash/find';
+import get from 'lodash/get';
+import cloneDeep from 'lodash/cloneDeep';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+
+interface TreeViewProps {
+    data: TreeNode[];
+    depth?: number;
+    deleteElement?: ReactElement;
+    getStyleClassCb?: (node, depth?) => {};
+
+    isCheckable?: (node, depth) => {};
+    isDeletable?: (node, depth) => {};
+    isExpandable?: (node, depth) => {};
+
+    keywordChildren?: string;
+    keywordChildrenLoading?: string;
+    keywordKey?: string;
+    keywordLabel?: string;
+
+    loadingElement?: ReactElement;
+    noChildrenAvailableMessage?: string;
+
+    onCheckToggleCb?: (arrayOfNodes, depth) => void;
+    onDeleteCb?: (node, updatedData, depth) => {};
+    onExpandToggleCb?: (node, depth) => void;
+    onUpdateCb?: (updatedData, depth) => void;
+
+    transitionEnterTimeout?: number;
+    transitionExitTimeout?: number;
+}
+
+interface TreeNode {
+    id: number;
+    name: string;
+    isExpanded: boolean;
+    isChecked: boolean;
+    children: TreeNode[];
+}
+
+export const TreeView: FunctionComponent<TreeViewProps> = (props: TreeViewProps): ReactElement => {
+
+    const [ treeData, setTreeData ] = useState<TreeNode[]>();
+    const [ lastCheckToggledNodeIndex, setLastCheckToggledNodeIndex ] = useState<number>();
+
+    useEffect(() => {
+        setTreeData(cloneDeep(props.data));
+    },[props.data]);
```

Comment:
It'll be better to destructure props at the begining of the component.

```
const { data } = props;
```

Reply:
Fixed with commit 3a6a06c